### PR TITLE
Feature 3222 docgeneration

### LINKED ISSUE #2906: “Event” printing function
### Situation Description
Sormas does not offer a print function. It would be interesting in that it allows information to be transmitted to a third party without giving them access as a user. The print function would be particularly interesting for tracking events and associated actions.

### Feature Description
Develop a function allowing users to print an HTML page from the information entered for each event:

- title of the event 

- event identifier

- event status

- event date

- description of the event

- Region, department and municipality of the event

- Community Contact Person

- For each action :

Title of the action
Description of the action
Creation date
Date of last modification
Status of the action

### Additional Information

@xavier-calland @bernardsilenou

### PR DESCRIPTION
**Pull request by syntakker**
Fixes #2906, #3222, #3480, #3991, #4057.
The fixes from #4095 are addressed in the refactoring.

### CODE REVIEW

File: sormas-backend/src/main/java/de/symeda/sormas/backend/docgeneration/EventDocumentFacadeEjb.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import de.symeda.sormas.api.docgeneneration.EventDocumentFacade;
+import de.symeda.sormas.api.event.EventParticipantFacade;
+import de.symeda.sormas.api.event.EventReferenceDto;
+
+public class EventDocumentFacadeEjb implements EventDocumentFacade {
+
+	private static final DocumentWorkflow DOCUMENT_WORKFLOW = DocumentWorkflow.EVENT_HANDOUT;
+
+	@EJB
+	DocumentTemplateFacade documentTemplateFacade;
```
Comment (StefanKock) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/docgeneration/EventDocumentFacadeEjb.java]: private modifier missing

**Comment by syntakker** [comment]
Feedback from review (@StefanKock):
- missing private modifer for EJBs, e.g., in `EventDocumentFacadeEjb`
- restrict EntityDtoHelper to `EntityDto` and `ReferenceDto` if possible
- custom path reset in `@Before` and `@After` methods in `AbstractDocGenerationTest`
- remove `try`/`catch` in `DocumentTemplatesView`

File: sormas-api/src/main/java/de/symeda/sormas/api/EntityDtoAccessHelper.java [hunk 2]
```diff
@@ -15,12 +30,13 @@
...
 
-	public static Object getPropertyValue(Object entity, String propertyKey) throws InvocationTargetException, IllegalAccessException {
+	public static Object getPropertyValue(HasUuid entity, String propertyKey) throws InvocationTargetException, IllegalAccessException {
 		if (entity == null) {
 			return null;
 		}
 		Class<?> entityClass = entity.getClass();
-		while (entityClass != null) {
+		while (entityClass != null
+			&& (EntityDto.class.isAssignableFrom(entity.getClass()) || ReferenceDto.class.isAssignableFrom(entity.getClass()))) {
```
Comment (StefanKock) [inline sormas-api/src/main/java/de/symeda/sormas/api/EntityDtoAccessHelper.java:39]: Why is this condition:
1. Not present on current development branch
2. Would needlessly reduce the functionality to only some implementations of `HasUuid`.

**Review by StefanKock — CHANGES_REQUESTED** [review]
A needless condition still present in `EntityDtoAccessHelper`

File: sormas-api/src/main/java/de/symeda/sormas/api/EntityDtoAccessHelper.java (hunk 2 above)
Comment (syntakker) [reply sormas-api/src/main/java/de/symeda/sormas/api/EntityDtoAccessHelper.java:39]: 1. in the current development branch, the superclass recursion is restricted by
```java
entityClass = HasUuid.class.isAssignableFrom(superclass) ? (Class<? extends HasUuid>) superclass : null;
```
which leads subsequently to breaking the `while` loop. This is in practice equivalent to the check for type `EntityDto` or `ReferenceDto` (see 2.)
2. Checking for type `HasUuid` is slightly more general than checking for `EntityDto` or `ReferenceDto`, however, the intended use case is `EntityDto` or `ReferenceDto`. Other implementations of `HasUuid` are either irrelevant (`AbstractDomainObject` is not used in this context) or are used in tests only. Thus the more restrictive check for `EntityDto` or `ReferenceDto` is actually more accurate here.

Comment (syntakker) [reply sormas-api/src/main/java/de/symeda/sormas/api/EntityDtoAccessHelper.java]: However, allowing for `HasUuid` does not hurt here. Removed obsolete checks, too.

**Pull request merged**